standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -3288,6 +3291,190 @@ public List<Void> run(List<String> input) throws Exception {
     return true;
   }
 
+  // a helper function which will firstly get the current COLUMN_STATS_ACCURATE parameter on table level
+  // secondly convert the JSON String into map, and update the information in it, and convert it back to JSON
+  // thirdly update the COLUMN_STATS_ACCURATE parameter with the new value on table level using directSql
+  public long updateColumnStatsAccurateForTable(Table table, List<String> droppedCols) throws MetaException {
+    String currentValue = table.getParameters().get("COLUMN_STATS_ACCURATE");
+    if (currentValue == null) return 0;
+
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+
+      // Deserialize the JSON into a map
+      Map<String, Object> statsMap = mapper.readValue(currentValue, new TypeReference<Map<String, Object>>() {});
+
+      // Get the COLUMN_STATS object if it exists
+      Object columnStatsObj = statsMap.get("COLUMN_STATS");
+
+      if (columnStatsObj instanceof Map) {
+        Map<String, String> columnStats = (Map<String, String>) columnStatsObj;
+
+        boolean removeAll = (droppedCols == null || droppedCols.isEmpty());
+
+        if (removeAll) {
+          // Remove entire column stats
+          statsMap.remove("COLUMN_STATS");
+        } else {
+          // Remove only the dropped columns
+          for (String col : droppedCols) {
+            columnStats.remove(col.toLowerCase());
+          }
+          if (columnStats.isEmpty()) {
+            statsMap.remove("COLUMN_STATS");
+          }
+        }
```

Comment:
I think you are trying to remove all column stats when either `droppedCols` is `null`, or when it's size is zero or equal to the size of the map `columnStats`, right?

In that case update `removeAll` to
```
boolean removeAll = droppedCols == null || droppedCols.isEmpty() || droppedCols.size() == columnStats.size();
```
You can then remove the block at the end:
```
          if (columnStats.isEmpty()) {
            statsMap.remove("COLUMN_STATS");
          }
```
This assumes that `List<String> droppedCols` has unique columns.